cdap-tms/src/main/java/co/cask/cdap/messaging/service/CoreMessagingService.java

```diff
@@ -221,22 +221,42 @@ protected void startUp() throws Exception {
     Queue<TopicId> asyncCreationTopics = new LinkedList<>();
 
     for (String topic : new HashSet<>(cConf.getTrimmedStringCollection(Constants.MessagingSystem.SYSTEM_TOPICS))) {
-      TopicId topicId;
-      try {
-        topicId = NamespaceId.SYSTEM.topic(topic);
-      } catch (IllegalArgumentException e) {
-        // Ignore invalid topic
-        LOG.warn("Ignore creation of invalid topic '{}'.", topic);
-        continue;
-      }
+      // Create topics with topic prefix and top partition
+      if (topic.contains(":")) {
+        String[] topicParts = topic.split(":");
+        int totalTopicNum;
+        if (topicParts.length != 2) {
+          LOG.warn("Ignore creation of invalid topic '{}'. Expecting topic in the format topicPrefix:numPartitions.",
+                   topic);
+          continue;
+        }
+        String topicPrefix = topicParts[0];
+        try {
+          NamespaceId.SYSTEM.topic(topicPrefix);
```

Comment:
If the topicPrefix is not valid, this call can catch it earlier without iterating through all the topics with numerical suffices of the same topicPrefix.